nomenclature/region_mapping_models.py

```diff
@@ -150,3 +150,34 @@ def create_from_region_mapping(cls, file: Union[Path, str]):
                 )
             mapping_input["common_regions"] = common_region_list
         return cls(**mapping_input)
+
+
+class ModelMappingCollisionError(ValueError):
+    template = "Model {model} is defined twice in {files}"
+
+    def __init__(
+        self, mapping1: RegionAggregationMapping, mapping2: RegionAggregationMapping
+    ) -> None:
+        files = (
+            mapping1.file.relative_to(Path.cwd()),
+            mapping2.file.relative_to(Path.cwd()),
+        )
+        super().__init__(self.template.format(model=mapping1.model, files=files))
+
+
+class RegionProcessor(BaseModel):
+    """Reads region mappings from a directory and uses them for region aggregation"""
```

Comment:
The existing text would be only for the `from_directory()` class-method...

```suggestion
    """Region aggregation mappings for scenario processing"""
```